CONTRIBUTING.md

```diff
@@ -0,0 +1,149 @@
+   - Clearly state that you are going to take on implementing it, if that's the case.
+2. Create a personal fork of the repository on GitHub (if you don't already have one).
+3. In your fork, create a branch off of main (if your change affects the current version) or the release-branch you want to contribute to (if you want to contribute to a future release).
+   - Name the branch according to our [branching guide](#branching-guide).
+   - Branches are useful since they isolate your changes from incoming changes from upstream. They also enable you to create multiple PRs from the same fork.
+4. Make and commit your changes to your branch.
+   - You can run static code checks b running `npm run lint`.
+   - You can run tests locally by running `npm run test:local:node`.
+   - Please follow our [Commit Messages](#commit-messages) guidance.
+5. Add new tests corresponding to your change, if applicable.
+6. Build the repository with your changes.
+   - Make sure that the builds are clean.
+   - Make sure that the tests are all passing, including your new tests.
+7. Make sure your branch has all changes from the main/release branch of the source repository.
+8. Increment the npm package version in the package.json (adhere to [semantic versioning](https://semver.org/)). Run `npm i` to make sure the change is propagated to package-lock.json.
+9. Create a pull request (PR) against the repository's **main** branch or the **release/\*** branch of the release you want to contribute to.
+   - State in the description what issue or improvement your change is addressing.
+   - Check if all the Continuous Integration checks are passing.
+10. Wait for feedback or approval of your changes from the team.
+11. When the PR is approved, and all checks are green, your PR will be merged.
+    - This will automatically trigger our continuous deployment pipeline, which builds and publishes a new version.
+    - You can delete the branch you used for making the change.
+
+### Branching Guide
+
+Branches must match the following pattern:
+
+```text
+<prefix>/[<issue-id>]<description>
+```
+
+- `<prefix>`: must be one of `chore`, `feature` or `bugfix`
+- `issue-id`: id of the GitHub issue created in step 1
+- `description`: concise description of the changes
+
+### Commit Messages
+
+Please format commit messages as follows:
+
+```text
+<prefix>: Summarize change in 65 characters or less
+
+Provide more detail after the first line. Leave one blank line below the
+summary and wrap all lines at 72 characters or less.
+
+If the change fixes an issue, leave another blank line after the final
+paragraph and indicate which issue is fixed in the specific format
+below.
+
+Fix #42
+```
+
+- `<prefix>`: must be one of `chore`, `feat`, `fix`, `refactor`, `test`
+
+Also do your best to factor commits appropriately, not too large with unrelated things in the same commit, and not too small with the same small change applied N times in N different commits.
+
+### PR - CI Process
```

Comment:
Es fehlt mir irgendwo ein "Chapter" bzgl. MIT Lizenz und wenn man Contributed, dann wird der Code auch MIT lizenzierbar contributed?

Reply:
Stand in dem MS-Guide auch nicht. Aber ist das nicht irgendwie logisch?

Reply:
Ne ist es nicht, afaik gibt es sogar Bots die bei PRs automatisch das Einverständnis einholen bzw. auch abfragen, ob es wirklich der eigene Code ist den man contributed - und nicht ggf. kopierter Code einer anderen Lizenz :(

Reply:
Hmm, ok. Meinst du es reicht dann, wenn wir das in den Contribution Guide schreiben?

Reply:
Wir können auch einfach SAPs https://cla-assistant.io/ nutzen, dann sind wir auf der sicheren Seite.

Reply:
Nochmal hier: aufgrund der [GitHub TOS](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#6-contributions-under-repository-license) ist das eigentlich unnötig. Ich würde es einfach lassen wie es ist.